File: website/Moe.La.Web/els-app/src/app/core/models/request.ts
import { SendingTypes } from './../enums/SendingTypes';
import { RequestStatus } from '../enums/RequestStatus';
import { RequestTypes } from '../enums/RequestTypes';
import { ChangeResearcherRequest } from './change-researcher-request';
import { KeyValuePairs } from './key-value-pairs';
import { RequestLetter } from './supporting-document-request';

export class RequestList {
  id:number=0; //رقم الطلب
  requestType!: KeyValuePairs; // نوع الطلب
  // نوع الطلب
  requestStatus!: KeyValuePairs; // حالة الطلب
  // حالة الطلب
  createdOn!: Date; // تاريخ تقديم الطلب
  // تاريخ تقديم الطلب
  lastTransactionDate: string=""; // تاريخ اخر حركة في الطلب
  caseId?: number; // رقم القضية
  createdBy!: KeyValuePairs; // مقدم الطلب : اسم مقدم الطلب
  // مقدم الطلب : اسم مقدم الطلب
  createdOnHigri?:string;
  lastTransactionDateHigri?:string;
}

export class Request {
  id!: number;
  caseId?: number;
  researcherId!: string;
  requestType!: string;
  note?: string;
  requestStatusId!: number;
  createdOn?: Date;
  changeResearcherRequest?: ChangeResearcherRequest;
}


export class RequestDto {
  id!: number;
  requestType!: RequestTypes;
  requestStatus!: RequestStatus;
  sendingType!: SendingTypes;
  receiverId?: string;
  receiverDepartmentId?: number;
  receiverRoleId?: string;
  relatedRequestId?: number;
  note?: string;
  createdBy?:KeyValuePairs;
  letter?: RequestLetter;

}

export class ObjectionPermitRequestDetails{

  id: number=0;
  caseId: number=0;
  request:RequestList;
  suggestedOpinon: KeyValuePairs = { id: 0, name: '' };
  note : string='';
  replyNote : string='';
}

File: services/src/Moe.La.Core/Entities/Requests/Request.cs
using Moe.La.Core.Enums;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace Moe.La.Core.Entities
{
    public class Request : BaseEntity<int>
    {
        /// <summary>
        /// نوع الطلب
        /// </summary>
        public RequestTypes RequestType { get; set; }

        /// <summary>
        /// حالة الطلب
        /// </summary>
        public RequestStatuses RequestStatus { get; set; }


        /// <summary>
        /// The sending type.
        /// </summary>
        public SendingTypes SendingType { get; set; }

        /// <summary>
        /// The request receiver user
        /// </summary>
        public Guid? ReceiverId { get; set; }

        public AppUser Receiver { get; set; }

        /// <summary>
        /// The request receiver general department id.
        /// </summary>
        public int? ReceiverBranchId { get; set; }

        /// <summary>
        /// The request receiver department id.
        /// </summary>
        public int? ReceiverDepartmentId { get; set; }

        /// <summary>
        /// The role to receive the request.
        /// </summary>
        public Guid? ReceiverRoleId { get; set; }

        /// <summary>
        /// The related request id.
        /// </summary>
        public int? RelatedRequestId { get; set; }

        public Request RelatedRequest { get; set; }

        /// <summary>
        /// سبب الطلب / ملاحظات
        /// </summary>
        public string Note { get; set; }

        /// <summary>
        /// للتصدير
        /// </summary>
        public bool IsExportable { get; set; }
        /// <summary>
        /// The user's id who created it.
        /// </summary>
        public Guid CreatedBy { get; set; }

        /// <summary>
        /// Navigation property to the user who created it.
        /// </summary>
        public AppUser CreatedByUser { get; set; }

        /// <summary>
        /// The user's id who updated it.
        /// </summary>
        public Guid? UpdatedBy { get; set; }

        /// <summary>
        /// Navigation property to the user who updated it.
        /// </summary>
        public AppUser UpdatedByUser { get; set; }

        /// <summary>
        /// The update datetime.
        /// </summary>
        public DateTime? UpdatedOn { get; set; }

        public ICollection<RequestTransaction> RequestTransactions { get; set; } = new Collection<RequestTransaction>();

        public ICollection<ConsultationSupportingDocumentRequest> ConsultationSupportingDocuments { get; set; } = new Collection<ConsultationSupportingDocumentRequest>();

        public RequestLetter Letter { get; set; }

    }
}
